tests/Integration/Tasks/CronTaskTest.php

```diff
@@ -503,11 +503,11 @@ public function prepareTaskNotRePrepareIfSuccessful()
 
         $pfTask->getFromDBByCrit(['name' => 'deploy']);
         $this->assertArrayHasKey('id', $pfTask->fields);
-        $pfTask->update([
-         'id'                      => $pfTask->fields['id'],
-         'reprepare_if_successful' => 0,
-         'is_active'               => 1
-        ]);
+
+        //update directly in DB to prevent reset jobstate
+        //see PluginGlpiinventoryTask->post_updateItem
+        $DB->query("UPDATE glpi_plugin_glpiinventory_tasks set `reprepare_if_successful` = 0 WHERE id = " . $pfTask->fields['id']);
```

Comment:
Please avoid using `DB::query()` directly; you can use `DB::update()` here.

Reply:
done !